tests/test.py

```diff
@@ -120,22 +98,97 @@ def has_json(self, condition_func: Callable[[dict], bool]) -> bool:
                 pass
         return False
 
-    def assert_json(self, condition_func: Callable[[dict], bool]) -> None:
-        if self.has_json(condition_func):
+    def assert_log(self, condition_func: Callable[[dict], bool]) -> None:
+        if self.has_log(condition_func):
             return
-        raise AssertionError("No JSON entry matched condition\n"
+        raise AssertionError("No entry matched condition\n"
                              "---- Log content ----\n"
                              f"{self.log_content}\n"
                              "-----------------")
 
-    def assert_not_json(self, condition_func: Callable[[dict], bool]) -> None:
-        if not self.has_json(condition_func):
+    def assert_not_log(self, condition_func: Callable[[dict], bool]) -> None:
+        if not self.has_log(condition_func):
             return
-        raise AssertionError("At least one JSON entry matched condition\n"
+        raise AssertionError("At least one entry matched condition\n"
                              "---- Log content ----\n"
                              f"{self.log_content}\n"
                              "-----------------")
 
+class FatraceRunnerText(FatraceRunnerAbstract):
+    def convert(self, line: str):
+        return parse_fatrace_text_line(line)
+
+class FatraceRunnerJson(FatraceRunnerAbstract):
+    def convert(self, line: str):
+        return json.loads(line)
+
+def parse_fatrace_text_line(line):
```

Comment:
FTR, I'm not a fan of this. This logic is way too complex for a test -- you would now need a whole unit test for *this* function.